fission/function.go

```diff
@@ -432,3 +433,69 @@ func fnPods(c *cli.Context) error {
 
 	return err
 }
+
+func fnInvoke(c *cli.Context) error {
+	// Temporary, switch once internal function routes are extracted to separate internal router.
+	routerURL := os.Getenv("FISSION_ROUTER")
+	if len(routerURL) == 0 {
+		fatal("Need FISSION_ROUTER set to your fission router.")
+	}
+
+	fnName := c.String("name")
+	if len(fnName) == 0 {
+		fatal("Need name of function, use --name")
+	}
+
+	url := fmt.Sprintf("http://%s/fission-function/%s", routerURL, fnName)
+
+	resp := httpRequest(c.String("method"), url, c.String("body"), c.StringSlice("header"))
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	fmt.Print(string(body))
+	defer resp.Body.Close()
+
+	return nil
+}
+
+func fnTest(c *cli.Context) error {
+	//client := getClient(c.GlobalString("server"))
+	fnName := c.String("name")
+
+	fnCreate(c)
+
+	routerURL := os.Getenv("FISSION_ROUTER")
+	if len(routerURL) == 0 {
+		fatal("Need FISSION_ROUTER set to your fission router.")
+	}
+
+	url := fmt.Sprintf("http://%s/fission-function/%s", routerURL, fnName)
+
+	maxRetries := 3
+	retryDelay := 100 * time.Millisecond
+	var resp *http.Response
+	fnReachable := false
+
+	for i := 0; i < maxRetries; i++ {
+		time.Sleep(retryDelay)
```

Comment:
No need to sleep before the first request?

Reply:
Instead of a fixed sleep, I am retrying. Would a fixed sleep be good idea?

Reply:
The retry is good, but the first request attempt can be sent without waiting (in other words the sleep should be after the request attempt)